--- cmovk.q.4652897	2020-12-30 21:19:16.530124066 +0800
+++ cmovk.q.zzzz	2020-12-30 20:54:41.439965292 +0800
@@ -1457,11 +1457,27 @@
 carry? a3 - 0
 a4 = z if !carry
 a5 = z if !carry
-a6 = z if !carry
+z = a6 if carry
 carry? a0 += a4
 carry? a1 += a5 + carry
 carry? a2 += a5 + carry
-       a3 += a6 + carry
+       a3 +=  z + carry
+
+t0 = a0
+t1 = a1
+t2 = a2
+h  = a3
+
+carry? a0 -= 18446744073709551597
+carry? a1 -= 18446744073709551615 - carry
+carry? a2 -= 18446744073709551615 - carry
+carry? a3 -= a6 - carry
+
+a0 = t0 if carry
+a1 = t1 if carry
+a2 = t2 if carry
+a3 = h  if carry
+
 
 t2 = stack_out
 
